pallets/struct-storage/src/lib.rs

```diff
@@ -1,4 +1,5 @@
 #![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::string_lit_as_bytes)]
```

Comment:
I think the better way to solve this is to go to https://github.com/paritytech/substrate/blob/ea501c2d61d27e7f0f76f93e5fa7e5c90882da2f/frame/support/procedural/src/storage/storage_struct.rs#L109 and use `LitByteStr` instead of `LitStr` for the storage prefix.